nifi-nar-bundles/nifi-email-bundle/nifi-email-processors/src/main/java/org/apache/nifi/processors/email/ListenSMTP.java

```diff
@@ -0,0 +1,383 @@
+        return propertyDescriptors;
+    }
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        this.relationships = Collections.unmodifiableSet(relationships);
+
+        final List<PropertyDescriptor> props = new ArrayList<>();
+        props.add(SMTP_PORT);
+        props.add(SMTP_HOSTNAME);
+        props.add(SMTP_MAXIMUM_CONNECTIONS);
+        props.add(SMTP_TIMEOUT);
+        props.add(SMTP_MAXIMUM_MSG_SIZE);
+        props.add(SSL_CONTEXT_SERVICE);
+        props.add(CLIENT_AUTH);
+        this.propertyDescriptors = Collections.unmodifiableList(props);
+
+    }
+
+    final ComponentLog logger = getLogger();
+
+    // Upon Schedule, reset the initialized state to false
+    @OnScheduled
+    public void onScheduled() {
+        initialized.set(false);
+    }
+
+    private synchronized void initializeSMTPServer(final ProcessContext context) throws Exception {
+        if (initialized.get()) {
+            return;
+        }
+
+        messages = new LinkedBlockingQueue<>(1024);
+        String clientAuth = null;
+
+        new SMTPMessageHandlerFactory(messages);
+
+        // If an SSLContextService was provided then create an SSLContext to pass down to the server
+        SSLContext sslContext = null;
+        final SSLContextService sslContextService = context.getProperty(SSL_CONTEXT_SERVICE).asControllerService(SSLContextService.class);
+        if (sslContextService != null) {
+            clientAuth = context.getProperty(CLIENT_AUTH).getValue();
+            sslContext = sslContextService.createSSLContext(SSLContextService.ClientAuth.valueOf(clientAuth));
+        }
+
+        final SSLContext finalSslContext = sslContext;
+        final String finalClientAuth = clientAuth;
+
+        SMTPMessageHandlerFactory smtpMessageHandlerFactory = new SMTPMessageHandlerFactory(messages);
+        final SMTPServer server = new SMTPServer(smtpMessageHandlerFactory) {
+
+            @Override
+            public SSLSocket createSSLSocket(Socket socket) throws IOException {
+                InetSocketAddress remoteAddress = (InetSocketAddress) socket.getRemoteSocketAddress();
+
+                SSLSocketFactory socketFactory = finalSslContext.getSocketFactory();
+
+                SSLSocket s = (SSLSocket) (socketFactory.createSocket(socket, remoteAddress.getHostName(), socket.getPort(), true));
+
+                s.setUseClientMode(false);
+
+
+                // For some reason the createSSLContext above is not enough to enforce
+                // client side auth
+                // If client auth is required...
+                if (finalClientAuth == "REQUIRED") {
+                    s.setNeedClientAuth(true);
+                }
+
+
+                return s;
+            }
+        };
+
+
+        // Set some parameters to our server
+        server.setSoftwareName("");
```

Comment:
I'm not sure how critical this software name is, however, at least I found this value is embedded in Headers:
```
Received: from 192.168.2.100 (localhost [127.0.0.1])
        by localhost
        with SMTP (SOFTWARE_NAME) id IPXNZWAU
        for xxxx@example.com;
        Mon, 27 Jun 2016 15:50:01 +0900 (JST)
...
```

It defaults to `"SubEthaSMTP " + Version.getSpecification()` inside the library code. "Apache NiFi" might be better than a blank string?